packages/template-chakra-storefront/src/components/with-translations/index.jsx

```diff
@@ -0,0 +1,122 @@
+// hoc/withTranslations.js
+import {useIntl} from 'react-intl'
+
+// Re-export defineMessages for a cohesive API. This allows components to import
+// everything they need for translations from a single file, simplifying imports
+// and making mocking for tests.
+// WARNING: you MUST keep the name "defineMessages" for react-intl static analysis to work.
+export {defineMessages} from 'react-intl'
+
+/**
+ * Higher-Order Component that abstracts React Intl complexity from components
+ * and provides a clean, magical messages API for internationalization.
+ * 
+ * WHY THIS HOC IS NEEDED:
+ * - Eliminates repetitive useIntl() and formatMessage() calls in every component
+ * - Keeps JSX clean and readable without i18n clutter
+ * - Abstracts React Intl complexity away from component logic
+ * - Provides type-safe message references instead of magic strings
+ * - Enables static analysis for message extraction while maintaining clean syntax
+ * 
+ * @param {React.Component} Component - The React component to enhance with translations
+ * @param {Object} messageDefinitions - Message definitions object created with defineMessages()
+ * 
+ * @returns {React.Component} Enhanced component that receives a magical `messages` prop
+ * 
+ * @example
+ * // 1. Define messages using defineMessages
+ * const userMessages = defineMessages({
+ *   welcome: {
+ *     id: 'user.welcome',
+ *     defaultMessage: 'Welcome, {name}!'
+ *   },
+ *   title: {
+ *     id: 'user.title', 
+ *     defaultMessage: 'User Dashboard'
+ *   }
+ * });
+ * 
+ * // 2. Create pure component (no i18n imports needed)
+ * const UserDashboard = ({ messages, user }) => (
+ *   <div>
+ *     <h1>{messages.title}</h1>
+ *     <p>{messages.welcome({ name: user.name })}</p>
+ *   </div>
+ * );
+ * 
+ * // 3. Wrap component with HOC
+ * export default withTranslations(UserDashboard, userMessages);
+ * 
+ * MAGICAL MESSAGES API:
+ * The `messages` prop provides a magical object where each message key works in multiple ways:
+ * 
+ * Static usage (no parameters needed):
+ *   {messages.title} → "User Dashboard"
+ * 
+ * Dynamic usage (with interpolation):
+ *   {messages.welcome({ name: 'John' })} → "Welcome, John!"
+ *   {messages.welcome.format({ name: 'John' })} → "Welcome, John!" (alternative syntax)
+ * 
+ * JavaScript usage (for alerts, confirmations, etc.):
+ *   const msg = messages.welcome({ name: user.name });
+ *   alert(msg);
+ * 
+ * HOW THE MAGIC WORKS:
+ * - Uses JavaScript Proxy to intercept property access
+ * - Each message becomes a hybrid object that's both a string and a function
```

Comment:
It's cool that it can both string and function. But translations don't need to be fancy IMO.

Reply:
second this!